src/fshtypes/FshCodeSystem.ts

```diff
@@ -27,8 +27,22 @@ export class FshCodeSystem extends FshEntity {
   }
 
   addConcept(newConcept: ConceptRule) {
-    if (this.rules.find(rule => rule instanceof ConceptRule && rule.code == newConcept.code)) {
-      throw new CodeSystemDuplicateCodeError(this.id, newConcept.code);
+    const existingConcept = this.rules.find(
+      rule => rule instanceof ConceptRule && rule.code == newConcept.code
+    ) as ConceptRule;
+    if (existingConcept) {
+      // if the new ConceptRule has only a code and a hierarchy,
+      // and the existing ConceptRule with that code has a matching hierarchy,
+      // the user may simply be referencing the existing concept to establish context.
+      if (
+        newConcept.display == null &&
+        newConcept.definition == null &&
+        isEqual(existingConcept.hierarchy, newConcept.hierarchy)
+      ) {
+        return;
+      } else {
+        throw new CodeSystemDuplicateCodeError(this.id, newConcept.code);
+      }
```

Comment:
Something like this:
```
CodeSystem: MyCS
* #foo
  * #bar
* #bar
```
counts as a duplicate, even though the first `bar` is a part of a hierarchy. Is that correct? It seems like you've considered this case and decided it is correct, but just want to double check.

Reply:
Yes, that is correct. Based on my understanding of [the csd-1 constraint](https://hl7.org/fhir/codesystem.html#invs), the above would not be allowed. When a concept in a CodeSystem is referenced, the reference doesn't contain any hierarchy information, so there would be no way to differentiate between the two uses of `#bar`.